sdk/ai/azure-ai-agents/azure/ai/agents/models/_patch.py

```diff
@@ -1538,21 +1568,154 @@ def add(self, tool: Tool):
         Add a tool to the tool set.
 
         :param Tool tool: The tool to add.
-        :raises ValueError: If a tool of the same type already exists.
+        :raises ValueError: If a tool of the same type already exists, or if an MCP tool with the same server label already exists.
```

Comment:
Pre-existing code, commenting here - the description of this class seem wrong. This is not an ABC?